File: tp-beer-frontend-typescript/src/interfaces/Beer.ts
import {KIND_OF_BEER} from "@/enums/enumBeers";

export interface Beer {
    id?: number;
    name: string;
    priceExcludingTaxCents: number;
    remark: string;
    alcoholDegrees: number;
    kindOfBeer: KIND_OF_BEER;
    owner: string;
}

export interface PageBeer {
    content: Beer[];
}

export interface SuccessResponse {
    data: Beer | Beer[];
    errors?: string[];
}

File: api-spring-boot/src/main/java/fr/noconsulting/restcrudapplication/entities/Beer.java
/**
 * Nom de classe : Beer
 * Description   : Classe de l'entité Beer
 * Date          : 13/05/2022
 * Copyright     : NoConsulting
 */

package fr.noconsulting.restcrudapplication.entities;

import lombok.*;

import javax.persistence.*;

@Entity
@Table(name = "beer") // Référence la table
@Getter @Setter @ToString @AllArgsConstructor @NoArgsConstructor
public class Beer {
    @Id // Référence la clé primaire de la table
    @GeneratedValue(strategy = GenerationType.IDENTITY) // Génération automatique par la bdd
    private int id;

    @Column(nullable = false, unique = true, length = 45)
    private String name;

    @Column()
    private int priceExcludingTaxCents;

    @Column(nullable = false, length = 400)
    private String remark;

    @Column()
    private int alcoholDegrees;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private KindOfBeer kindOfBeer;

    @Column(nullable = false, length = 45)
    private String owner;
}
